component/src/main/java/org/ballerinalang/net/kafka/nativeimpl/functions/consumer/GetBeginningOffsets.java

```diff
@@ -0,0 +1,122 @@
+        BStruct consumerStruct = (BStruct) getRefArgument(context, 0);
+        KafkaConsumer<byte[], byte[]> kafkaConsumer = (KafkaConsumer) consumerStruct
+                .getNativeData(Constants.NATIVE_CONSUMER);
+        if (kafkaConsumer == null) {
+            throw new BallerinaException("Kafka Consumer has not been initialized properly.");
+        }
+
+        BRefValueArray partitions = ((BRefValueArray) getRefArgument(context, 1));
+        ArrayList<TopicPartition> partitionList = new ArrayList<TopicPartition>();
+
+        for (int counter = 0; counter < partitions.size(); counter++) {
+            BStruct partition = (BStruct) partitions.get(counter);
+            String topic = partition.getStringField(0);
+            int partitionValue = new Long(partition.getIntField(0)).intValue();
+            partitionList.add(new TopicPartition(topic, partitionValue));
+        }
+
+        try {
+            Map<TopicPartition, Long> offsetMap = kafkaConsumer.beginningOffsets(partitionList);
+            List<BStruct> offsetList = new ArrayList<>();
+            if (!offsetMap.entrySet().isEmpty()) {
+                offsetMap.entrySet().forEach(offset -> {
+                    BStruct offsetStruct = createOffsetStruct(context);
+                    BStruct partitionStruct = createPartitionStruct(context);
+                    partitionStruct.setStringField(0, offset.getKey().topic());
+                    partitionStruct.setIntField(0, offset.getKey().partition());
+                    offsetStruct.setRefField(0, partitionStruct);
+                    offsetStruct.setIntField(0, offset.getValue());
+                    offsetList.add(offsetStruct);
+                });
+            }
+            return getBValues(new BRefValueArray(offsetList.toArray(new BRefType[0]),
+                    createOffsetStruct(context).getType()));
+        } catch (KafkaException e) {
+            return getBValues(null, BLangVMErrors.createError(context, 0, e.getMessage()));
+        }
+    }
+
+    private BStruct createOffsetStruct(Context context) {
+        PackageInfo kafkaPackageInfo = context.getProgramFile()
+                .getPackageInfo(Constants.KAFKA_NATIVE_PACKAGE);
+        StructInfo consumerRecordStructInfo = kafkaPackageInfo.getStructInfo(Constants.OFFSET_STRUCT_NAME);
+        BStructType structType = consumerRecordStructInfo.getType();
+        BStruct bStruct = new BStruct(structType);
+        return bStruct;
+    }
+
+    private BStruct createPartitionStruct(Context context) {
+        PackageInfo kafkaPackageInfo = context.getProgramFile()
+                .getPackageInfo(Constants.KAFKA_NATIVE_PACKAGE);
+        StructInfo consumerRecordStructInfo = kafkaPackageInfo
+                .getStructInfo(Constants.TOPIC_PARTITION_STRUCT_NAME);
+        BStructType structType = consumerRecordStructInfo.getType();
+        BStruct bStruct = new BStruct(structType);
+        return bStruct;
+    }
```

Comment:
Instead of having different methods (createPartitionStruct,createRecordStruct,createOffsetStruct) over different classes lets have a common method createStruct(Context context, String structname) in a common place

Reply:
+1